brain/task_executor.py

```
@@ -232,14 +235,27 @@ async def plugin_list_provider(self, force_refresh: bool = True) -> List[Dict[st
-        return AsyncOpenAI(
-            api_key=api_config['api_key'],
-            base_url=api_config['base_url'],
-            max_retries=0
-        )
+        key = (api_config['api_key'], api_config['base_url'])
+
+        if self._cached_client_key != key:
+            # 配置变更，关闭旧 client 并创建新实例
+            if self._cached_client is not None:
+                try:
+                    asyncio.ensure_future(self._cached_client.close())
```

Ruff (0.15.9)

[warning] Store a reference to the return value of `asyncio.ensure_future`

(RUF006)